src/app/log-updates.directive.ts

```diff
@@ -1,12 +1,20 @@
-import { Directive, ElementRef } from '@angular/core';
+import { Directive, ElementRef, HostListener, OnChanges, SimpleChanges } from '@angular/core';
 
 @Directive({
   selector: '[appLogUpdates]',
 })
-export class LogUpdatesDirective {
+export class LogUpdatesDirective implements OnChanges {
   constructor(private elemRef: ElementRef) {
     console.log(`Attached to`, this.elemRef.nativeElement);
   }
 
-  // TODO: Log name update events here
+  ngOnChanges(changes: SimpleChanges) {
+    // works as well
+    // console.log('Hello ' + changes.log.currentValue);
```

Comment:
This only works because there is also an input binding but without it it will not work =)
Good find though, directives can access same inputs as components!